cmd/juju/cloud/add_test.go

```diff
@@ -757,3 +706,260 @@ func (s *addSuite) TestInvalidCredentialMessage(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(cmdtesting.Stderr(ctx), jc.Contains, "Cloud credential is not accepted by cloud provider: running test")
 }
+
+type addOpenStackSuite struct {
+	jujutesting.IsolationSuite
+}
+
+var _ = gc.Suite(&addOpenStackSuite{})
+
+func (s *addOpenStackSuite) TearDownTest(c *gc.C) {
+	s.IsolationSuite.TearDownTest(c)
+	os.Unsetenv("OS_CACERT")
+	os.Unsetenv("OS_AUTH_URL")
+}
+
+func (*addOpenStackSuite) TestInteractiveOpenstackNoCloudCert(c *gc.C) {
+	myOpenstack := cloudfile.Cloud{
+		Name:      "os1",
+		Type:      "openstack",
+		AuthTypes: []cloudfile.AuthType{"userpass", "access-key"},
+		Endpoint:  "http://myopenstack",
+		Regions: []cloudfile.Region{
+			{
+				Name:     "regionone",
+				Endpoint: "http://boston/1.0",
+			},
+		},
+	}
+
+	var expectedYAMLarg = "" +
+		"auth-types:\n" +
+		"- userpass\n" +
+		"- access-key\n" +
+		"certfilename: \"\"\n" +
+		"endpoint: http://myopenstack\n" +
+		"regions:\n" +
+		"  regionone:\n" +
+		"    endpoint: http://boston/1.0\n"
+
+	var input = "" +
+		/* Select cloud type: */ "openstack\n" +
+		/* Enter a name for your openstack cloud: */ "os1\n" +
+		/* Enter the API endpoint url for the cloud []: */ "http://myopenstack\n" +
+		/* Enter the filename of the CA certificate to access OpenStack cloud (optional) [] */ "\n" +
```

Comment:
I know these are the tests but wording suggestion

    If your cloud requires a CA certificate please provide the path to the certificate (optional) [none]:

Reply:
There isn't a good way to get rid of a space at the beginning:
"Enter , if required, the path to a CA"... 

Sorry I wasn't more specific, but "Enter " gets tacked to the beginning on the string I provide..

Does "Enter if required, the path to a CA" work for you?

Reply:
no, I don't think that's proper punctuation there. #$@$R#@ this code is here just to irritate me I'm beginning to think.

Reply:
Enter a path to the CA certificate for your cloud if one is required to access it. (optional) [none]

I hate that I can't think of a better way to get the optional-ity of the question towards the front of the line given the stupid constraints of this code.

Reply:
updated.